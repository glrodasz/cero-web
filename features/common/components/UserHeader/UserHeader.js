import React from 'react'
import PropTypes from 'prop-types'
import { Avatar, Spacer, Heading, Paragraph } from '@glrodasz/components'

import styles from './UserHeader.module.css'

const UserHeader = ({ avatar, title, text, isPrimary }) => {
  return (
    <div className={styles['user-header']}>
      <Avatar src={avatar} />
      <Spacer.Horizontal size="sm" />
      <div className={styles['text']}>
        <Heading size="lg">{title}</Heading>
        {isPrimary ? (
          <Heading color="primary">{text}</Heading>
        ) : (
          <Paragraph size="lg">{text}</Paragraph>
        )}
      </div>
    </div>
  )
}

UserHeader.propTypes = {
  avatar: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  isPrimary: PropTypes.bool,
}

export default UserHeader
