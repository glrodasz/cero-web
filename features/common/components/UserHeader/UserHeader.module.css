.user-header {
  display: flex;
}

.text {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}
