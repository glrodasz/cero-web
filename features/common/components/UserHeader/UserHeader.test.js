import UserHeader from './UserHeader'
import { render } from '@testing-library/react'

jest.mock('@glrodasz/components', () => {
  const { dummyRender } = require('../../../../utils/testing')

  return {
    Avatar: dummyRender('Avatar'),
    Spacer: { Vertical: dummyRender('Space.Vertical') },
    Heading: dummyRender('Heading'),
    Paragraph: dummyRender('Paragraph'),
  }
})

describe('[ features / common / components / UserHeader ]', () => {
  describe('when `UserHeader` is mounted', () => {
    it('should render', () => {
      const props = {
        avatar: 'avatar',
        title: 'title',
        text: 'text',
      }
      const { asFragment } = render(<UserHeader {...props} />)
      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe('when `UserHeader` is mounted with `isPrimary` as `true`', () => {
    it('should render', () => {
      const props = {
        avatar: 'avatar',
        title: 'title',
        text: 'text',
        isPrimary: true,
      }
      const { asFragment } = render(<UserHeader {...props} />)
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
