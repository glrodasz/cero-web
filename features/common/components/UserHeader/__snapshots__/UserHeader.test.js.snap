// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ features / common / components / UserHeader ] when \`UserHeader\` is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="user-header"
  >
    [Avatar {"src":"avatar"} /] [Space.Vertical {"size":"sm"} /] 
    <div
      class="text"
    >
      [Heading {"size":"lg"}]title[/Heading] [Paragraph {"size":"lg"}]text[/Paragraph] 
    </div>
  </div>
</DocumentFragment>
`;

exports[`[ features / common / components / UserHeader ] when \`UserHeader\` is mounted with \`isPrimary\` as \`true\` should render 1`] = `
<DocumentFragment>
  <div
    class="user-header"
  >
    [Avatar {"src":"avatar"} /] [Space.Vertical {"size":"sm"} /] 
    <div
      class="text"
    >
      [Heading {"size":"lg"}]title[/Heading] [Heading {"color":"primary"}]text[/Heading] 
    </div>
  </div>
</DocumentFragment>
`;
